### ISSUE TITLE
feat: implement multiple config store options

### ISSUE THREAD

**Issue by dalmaer** [polymath-ai/polymath-ai#10]
E.g. load from host.SECRET.json, env variables, Firestore, etc.

**Comment by dglazkov** [comment]
I have a really neat idea of references that I want to try:

`polymath.config`:
```json
{
   "references": "firebase:<url>"
}
```

**Comment by dalmaer** [comment]
so `polymath.config` is the hub that links off to `firebase:...` or `file:...` or `supabase:...` etc?

**Comment by dglazkov** [comment]
yeah, I am still trying to figure this out, but the way `tsconfig.json` can use references is giving me all kinds of ideas. What if our config can reference things by a URI?
